File: src/tools/visualStates_py/codegen/cpp/temporaltransition.cpp
#include "temporaltransition.h"
#include <sys/time.h>

TemporalTransition::TemporalTransition(int id, int destinationId, int elapsedTime):Transition(id, destinationId) {
    this->elapsedTime = elapsedTime;
}

void TemporalTransition::init() {
    startTime = getCurrentTime();
}

bool TemporalTransition::checkCondition() {
    long diffTime = getCurrentTime()-startTime;
    if (diffTime > elapsedTime) {
        return true;
    } else {
        return false;
    }
}

long TemporalTransition::getCurrentTime() {
    struct timeval a;
    gettimeofday(&a, 0);
    return (a.tv_sec * 1000000 + a.tv_usec)/1000;
}

File: src/tools/visualStates_py/codegen/python/temporaltransition.py
from gui.codegen.transition import Transition
from time import time

class TemporalTransition(Transition):
    def __init__(self, id, destinationId, elapsedTime):
        super().__init__(id, destinationId)
        # elapsed time in milliseconds
        self.elapsedTime = elapsedTime
        self.startTime = None

    def init(self):
        self.startTime = time()*1000

    def runCode(self):
        pass

    def checkCondition(self):
        super().checkCondition()
        diffTime = (time()*1000)-self.startTime
        if diffTime > self.elapsedTime:
            return True
        else:
            return False
